xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -607,8 +607,36 @@ void CLinuxRendererGLES::RenderUpdateVideo(bool clear, DWORD flags, DWORD alpha)
     CDVDVideoCodecIMXBuffer *buffer = m_buffers[m_iYV12RenderBuffer].IMXBuffer;
     if (buffer != NULL && buffer->IsValid())
     {
+      // this hack is needed to get the 2D mode of a 3D movie going
+      if (g_graphicsContext.GetStereoMode())
+        g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_LEFT);
+
       ManageDisplay();
-      g_IMXContext.SetBlitRects(m_sourceRect, m_destRect);
+
+      if (g_graphicsContext.GetStereoMode())
+        g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_OFF);
+
+      CRect dstRect(m_destRect);
+      CRect srcRect(m_sourceRect);
+      RENDER_STEREO_MODE stereo_mode = g_graphicsContext.GetStereoMode();
```

Comment:
Yep. Will do. Thx.

2015-05-28 7:34 GMT+02:00 Chris Browet <notifications@github.com>:

> In xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
> <https://github.com/xbmc/xbmc/pull/7186#discussion_r31204401>:
>
> >        ManageDisplay();
> > -      g_IMXContext.SetBlitRects(m_sourceRect, m_destRect);
> > +
> > +      if (g_graphicsContext.GetStereoMode())
> > +        g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_OFF);
> > +
> > +      CRect dstRect(m_destRect);
> > +      CRect srcRect(m_sourceRect);
> > +      RENDER_STEREO_MODE stereo_mode = g_graphicsContext.GetStereoMode();
>
> Sorry, one thing: I'd move "stereo_mode =" up and use it rather than call
> "GetStereoMode" 3 times
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7186/files#r31204401>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B